include/aos/common/pkcs11/pkcs11.hpp

```diff
@@ -0,0 +1,743 @@
+        (void)allocator;
+    }
+
+    /**
+     * Creates an RSA key pair on the token.
+     *
+     * @param id key pair id.
+     * @param label key pair label.
+     * @param bitsCount key size.
+     * @return RetWithError<PrivateKey>.
+     */
+    RetWithError<PrivateKey> GenerateRSAKeyPairWithLabel(
+        const Array<uint8_t>& id, const String& label, size_t bitsCount)
+    {
+        (void)id;
+        (void)label;
+        (void)bitsCount;
+
+        return {PrivateKey()};
+    }
+
+    /**
+     * Creates a ECDSA key pair on the token using curve.
+     *
+     * @param id token id.
+     * @param label label.
+     * @param curve elliptic curve.
+     * @return RetWithError<PrivateKey>.
+     */
+    RetWithError<PrivateKey> GenerateECDSAKeyPairWithLabel(
+        const Array<uint8_t>& id, const String& label, EllipticCurve curve)
+    {
+        (void)id;
+        (void)label;
+        (void)curve;
+
+        return {PrivateKey()};
+    }
+
+    /**
+     * Retrieves a previously created asymmetric key pair.
+     *
+     * @param id key id.
+     * @param label key label.
+     * @return RetWithError<PrivateKey>.
+     */
+    RetWithError<PrivateKey> FindPrivateKey(const Array<uint8_t>& id, const String& label)
+    {
+        (void)id;
+        (void)label;
+
+        return {PrivateKey()};
+    }
+
+    /**
+     * Deletes private key for the token.
+     *
+     * @param key private key to be deleted.
+     * @return Error.
+     */
+    Error DeletePrivateKey(const PrivateKey& key)
+    {
+        (void)key;
+
+        return {};
+    }
+
+    /**
+     * Imports a certificate onto the token.
+     *
+     * @param id certificate id.
+     * @param label certificate label.
+     * @param cert certificate.
+     * @return Error.
+     */
+    Error ImportCertificate(const Array<uint8_t>& id, const String& label, const crypto::x509::Certificate& cert)
+    {
+        (void)id;
+        (void)label;
+        (void)cert;
+
+        return {};
+    }
+
+    /**
+     * Checks whether certificate with specified attributes has been imported.
+     *
+     * @param issuer certificate issuer.
+     * @param serialNumber certificate serialNumber.
+     * @return Error.
+     */
+    RetWithError<bool> HasCertificate(const Array<uint8_t>& issuer, const Array<uint8_t>& serialNumber)
+    {
+        (void)issuer;
+        (void)serialNumber;
+
+        return {false};
+    }
+
+    /**
+     * Deletes a previously imported certificate.
+     *
+     * @param id certificate id.
+     * @param label certificate label.
+     * @return Error.
+     */
+    Error DeleteCertificate(const Array<uint8_t>& id, const String& label)
+    {
+        (void)id;
+        (void)label;
+
+        return {};
+    }
+
+    /**
+     * Converts PKCS11 byte array to string.
+     *
+     * @param src source byte array.
+     * @param dst destination string.
+     * @return Error.
+     */
+    static Error Convert(const Array<uint8_t>& src, String& dst)
```

Comment:
I've asked before but didn't find the answer. What function suppose to do? Just copy byte array to string?
If so, it is not task for pkcs11.  It should be done in common.

Reply:
here is an example from documentation
slotDescription       character-string description of the slot.  MUST be padded with the blank character (‘ ‘).  MUST NOT be null-terminated.